core/src/main/java/hudson/model/Queue.java

```diff
@@ -336,9 +341,9 @@ public Void call() throws Exception {
         }
     });
 
-    private transient final Lock lock = new ReentrantLock();
+    /*really just for Executor*/ transient final Lock lock = new ReentrantLock();
```

Comment:
except it's really `ResourceController.lock` that Executor wants so we have gone from 1 lock to 2 locks and a mutator thread... I am not seeing how this helps